### ISSUE TITLE
"Request failed with status code Forbidden" on start up

### ISSUE THREAD

**Issue by kupet00** [robvdpol/RaceControl#414]
**Describe the bug**
When starting the app, I get greeted by the following login screen ("Request failed with status code Forbidden"):
![f1rc](https://user-images.githubusercontent.com/40399761/162154648-f04be1af-753d-45f5-b986-e9b6e751bdbc.png)

Logging in on the official website or Android App does work with the same credentials.

**To Reproduce**
1. Start the app

**Expected behavior**
Either a login screen without errors and being able to login in or no login screen at all (being automatically logged in)

**Desktop (please complete the following information):**
 - Windows 11 Pro 21H2
 - 64bit
 - App version 2.6.3

**F1TV account**
- Subscription type: Pro
- Netherlands
- Using VPN or proxy? No

**Comment by PostMalone** [comment]
I have the same issue as you. Request failed with status code Forbidden

Its happening hours before FP1 in Australia.

**Comment by robvdpol** [comment]
Hmm, they might have disabled the api key we're using for the login request. We would have to check the contents of the login request from one of the official apps to see what has changed. I'm not sure how to do that unfortunately :/ Maybe someone else is able to do it. It's a bit tricky since it's of course encrypted.

**Comment by sartsj** [comment]
API key seems fine, what I needed to do to make the auth request working again was adding this new euconsent cookie (not going to paste it here, not sure what it contains).

**Comment by robvdpol** [comment]
@sartsj The cookie shouldn't contain private information, you could try to decode it at http://iabtcf.com/#/decode to verify. An would really like to have it so I can fix the login.

**Comment by sartsj** [comment]
Specifically, the cookie is the one called 'reese84'

**Comment by robvdpol** [comment]
Ah the reese84 stuff again. There used to be an exception for 'RaceControl' user agent not needing it, but apparantly they removed that exception :( They seem to be cracking down on third party tools this season.

See also https://github.com/SoMuchForSubtlety/f1viewer/issues/111

**Comment by SoMuchForSubtlety** [comment]
Maybe we can send them a nice email asking to add the exception back :smile:

**Comment by dhuit90** [comment]
This cookie is associated with Imperva Networks and is used to determine if a user will receive a captcha.

**Comment by dhuit90** [comment]
https://github.com/SoMuchForSubtlety/f1viewer/issues/231#issuecomment-1091454437

This gives us some hope

**Comment by dhuit90** [comment]
@sartsj could you share your postman request? I'm having no luck logging in, even after syncing the reese84 cookie from my browser. Still get the "Pardon Our Interruption
As you were browsing something about your browser made us think you were a bot. There are a few reasons this might happen..."  403 page. 

Using the correct cookie, api key, user-agent and credentials

**Comment by sartsj** [comment]
@dhuit90 I'm honestly not having much luck anymore either. The only way I can get a succesfull auth request in postman now is by literally copying the one from my firefox dev console after logging in on the f1tv website.

Even if I only copy the reese84 cookie value from that request I can't get it to work anymore. I must be messing something up.

**Comment by dhuit90** [comment]
So one (very ugly) solution **_might_** be running a headless browser to get the token and use that to handle the login in RaceControl... I guess

**Comment by robvdpol** [comment]
> I'll see how far [this ](https://documenter.getpostman.com/view/11462073/TzY68Dsi)gets me

The reese84-token that you get from that request doesn't work. When I use the reese84-token from my webbrowser console it does work.

**Comment by dhuit90** [comment]
> > I'll see how far [this ](https://documenter.getpostman.com/view/11462073/TzY68Dsi)gets me
> 
> The reese84-token that you get from that request doesn't work. When I use the reese84-token from my webbrowser console it does work.

Same here unfortunately, i've been syncing the cookies from my browser (logged in to F1TV website) to postman but no luck

**Comment by MueR** [comment]
This curl request will get you the required value for a reese84 cookie. Still unsure how it works though.
```curl --location --request POST 'https://api.formula1.com/6657193977244c13?d=account.formula1.com' \
--data-raw '{"solution":{"interrogation":{"st":162229509,"sr":1959639815,"cr":78830557},"version":"stable"},"error":null,"performance":{"interrogation":185}}'```

**Comment by dhuit90** [comment]
@flyingwraptor is the code you supplied in https://github.com/SoMuchForSubtlety/f1viewer/issues/111#issuecomment-727553895 a solution for this?

**Comment by eXhumer** [comment]
I have a better solution for subscription token handling, but it requires access to an embedded web browser from which you can listen to cookie events. I used Qt's Framework in C++ to essentially listen for `login-session` session cookie for formula1.com in a Qt WebEngine after allowing a user to log in. Upon successful login, F1 will add the cookie which includes the subscription token.

https://github.com/eXhumer/eXF1TV/blob/main/Source/F1TVService.cxx#L30

**Comment by robvdpol** [comment]
> 

I was thinking of doing something similar using an Edge WebView2 component. Let the user login on the official website and fetch the sessiontoken from response or cookie.

**Comment by eXhumer** [comment]
You can also allow manual token as temporary fix this weekend. Here is an easy way to get the current logged in session token from browser.
![image](https://user-images.githubusercontent.com/62310242/162210203-414840b2-5d3f-4785-ae0b-bef8debf0603.png)

**Comment by koningcool** [comment]
Is there any easy actual working fix for this right now?

**Comment by robvdpol** [comment]
Nope but I'm working on a fix. Proof of concept is ready, just need to integrate it into Race Control now. Hope I can do it tonight.

**Comment by BeanXOfficial** [comment]
> Owner

You're a legend, thanks for all the effort put into this project!

**Comment by BeanXOfficial** [comment]
> Nope but I'm working on a fix. Proof of concept is ready, just need to integrate it into Race Control now. Hope I can do it tonight.

You're a legend, thanks for all the work you put into this project!

**Comment by Tanguille** [comment]
> You can also allow manual token as temporary fix this weekend. Here is an easy way to get the current logged in session token from browser. ![image](https://user-images.githubusercontent.com/62310242/162210203-414840b2-5d3f-4785-ae0b-bef8debf0603.png)

Nice idea. Do you use that token instead of your password then? 
Is there a way to do this via the android app? Can only log in there. In all my browsers I can't login. After some research I discovered this could be because F1TV is restricting IP-adresses to avoid account sharing. Any way I could fix this? 
Answers to my questions would be much appreciated. Or if robvdpol could achieve a miracle and fix RaceControl until FP1 it would solve all of my above questions. Good luck and courage and thanks in advance!

**Comment by thedave42** [comment]
> I have a better solution for subscription token handling, but it requires access to an embedded web browser from which you can listen to cookie events. I used Qt's Framework in C++ to essentially listen for `login-session` session cookie for formula1.com in a Qt WebEngine after allowing a user to log in. Upon successful login, F1 will add the cookie which includes the subscription token.
> 
> https://github.com/eXhumer/eXF1TV/blob/main/Source/F1TVService.cxx#L30

[This is what I'm doing as well](https://github.com/thedave42/f1tv-dl/blob/56d88e517d8b2548b367b25ef45451652cf44a3e/lib/f1tv-token.js).   I'm using Chromium and there is still a need for some extensions to get around the checks.

**Comment by AtlAntA118** [comment]
This might not be very usefull at all but i figured i might share this bit of info anyway. I can login just fine with RaceControl 2.6.3. No error message whatsoever and i can play all streams just fine.

The thing i do run on my PC is adguard with the extension called 'i dont care about cookies'. This *might* be doing something with the cookies and how they are handled.

**Comment by koningcool** [comment]
Did you install i dont care about cookies as a extension in your browser?

**Comment by AtlAntA118** [comment]
> Did you install i dont care about cookies as a extension in your browser?

No i run the desktop app of adguard and i use the 'i dont care about cookies' list there. 
![image](https://user-images.githubusercontent.com/35236700/162283288-06f7c8cc-8516-419c-b5fa-ad563526abdf.png)

Im not going to pretend i know how cookies work or are being handled but i think this might make a difference opposed to a webbrowser based adblocker/cookiehandler.

**Comment by anthonylavado** [comment]
@AtlAntA118 Thanks for sharing that. It might be unrelated though. It looks like if you're _already_ logged in, your session token is still valid for a little while. If that token has expired, or you sign out/in, then you run in to the issue seen here.

**Comment by robvdpol** [comment]
I have a solution but when I run it in release mode it doesn't work somehow :( I have to look into it tomorrow.

**Comment by eXhumer** [comment]
The "I don't care about cookies" is a browser extension / add-on. Nothing to do with application cookie handling.

**Comment by SkYNewZ** [comment]
> I have a solution but when I run it in release mode it doesn't work somehow :( I have to look into it tomorrow.

Good luck to you, thanks for your work and have a good Australian GP!

**Comment by MatasVin** [comment]
cant wait for this to be fixed, i guess the web version will suffice for now

**Comment by Jurtaani** [comment]
i was able to login and continue using this app just fine, and the previous races from this and older seasons wor just fine.
but i did notice that i was not able to watch Australian gp fp1 live or now as a archived session.
other players stay empty but the internal one tries to load something, however that is just some pink mess with no audio.
have they changed some drm things or is this just related to this issue as well and the upcoming 2.6.4 will resolve this issue as well?
![2022-04-08 (1)](https://user-images.githubusercontent.com/96973990/162370335-94b763a1-9c6e-4baf-97dc-de0319aa0b02.png)

**Comment by eXhumer** [comment]
I fear this isn't something any update can fix. F1 started to return Widevine'd streams for anything that was live or moved into archive recently.

**Comment by PostMalone** [comment]
This is becoming hell... F1tv is not messing around anymore. They only want to achieve one goal and that is to prevent any 3rd party tools running

**Comment by robvdpol** [comment]
Ok, this will require some more work. It basically means abandoning the Flyleaf video library and replacing it with an Edge WebView2 browser control that runs VideoJS. I have messed around with this in the past, and it should be able to support Widevine DRM. But it will take some time to implement, so please be patient.

**Comment by aeleos** [comment]
> Ok, this will require some more work. It basically means abandoning the Flyleaf video library and replacing it with an Edge WebView2 browser control that runs VideoJS. I have messed around with this in the past, and it should be able to support Widevine DRM. But it will take some time to implement, so please be patient.

Take your time, and thanks for all of the work you do

**Comment by scns** [comment]
> 

Hope you can find a fix, and tnx for all the work you do.  It's an amazing tool!

**Comment by scns** [comment]
> 



> Ok, this will require some more work. It basically means abandoning the Flyleaf video library and replacing it with an Edge WebView2 browser control that runs VideoJS. I have messed around with this in the past, and it should be able to support Widevine DRM. But it will take some time to implement, so please be patient.

Hope you can find a fix, and tnx for all the work you do. It's an amazing tool!

**Comment by MatasVin** [comment]
Is there a chance to fix this issue before the australian gp race session?

**Comment by Jurtaani** [comment]
> Ok, this will require some more work. It basically means abandoning the Flyleaf video library and replacing it with an Edge WebView2 browser control that runs VideoJS. I have messed around with this in the past, and it should be able to support Widevine DRM. But it will take some time to implement, so please be patient.

Fingers crossed that this doesn't make the great performance of the current method go away.
i mean i can easily ran the four streams layout on the archived races without my laptop fans even noticing anything.
but soon as i go and put up two browsers to at least get the F1live+one driver camera layout.
it makes my laptop fans go crazy.

**Comment by hangro** [comment]
> > Is there a chance to fix this issue before the australian gp race session?
> 
> Nope, unfortunately not.

Succes Rob, and try to enjoy the Australian GP...

**Comment by MueR** [comment]
Get in touch with F1TV Support and complain (politely) about the changes to this DRM. Maybe, just maybe, they'll think about it again, or give 3rd party developers like Rob a headsup for the next change.

Rob, thanks for your hard work man.

**Comment by yukikaze04** [comment]
I told them I'm not renewing my subscription if RaceControl won't be allowed to work. I'd rather watch F1 for free than pay to use their joke of a website.

**Comment by PostMalone** [comment]
> I told them I'm not renewing my subscription if RaceControl won't be allowed to work. I'd rather watch F1 for free than pay to use their joke of a website.

I requested a refund on my annual Sub. As a reason i stated ban on 3rd party apps that made f1tv great. Now its total garbage with their crap Web Player, not to mention the cpu load when you open a few tabs.

**Comment by yukikaze04** [comment]
The web player is a disaster, it crashes for me every ~30 minutes, like clockwork. Doesn't matter which browser I use. It's been happening for like at least a year now, but I didn't care cause we had RaceControl. Now I care.

**Comment by dhuit90** [comment]
> > I told them I'm not renewing my subscription if RaceControl won't be allowed to work. I'd rather watch F1 for free than pay to use their joke of a website.
> 
> I requested a refund on my annual Sub. As a reason i stated ban on 3rd party apps that made f1tv great. Now its total garbage with their crap Web Player, not to mention the cpu load when you open a few tabs.

Good idea, I'm doing this too. I surely won't get a refund but it might make them reconsider their decisions regarding 3rd party apps

**Comment by Gedioa** [comment]
I am also going to tell them i will nor renew my moth sub. Reasons: There web player sucks. I almost never watch the F1 live, now when i log-in to the f1 site you  see who won the race, so there is no point in watching the race!
Will also tell them I will start using illegaal streams!

**Comment by moi427** [comment]
Of course the same problem here. I also sent them a similar mail to allow RaceControl access, hope they get these a lot.

Thanks a lot for your fabulous work Rob!!

**Comment by Ainson** [comment]
Thanks for working on a fix.

I suspect this is related to a poll that was recently on F1 Fan Voice about F1TV where for sure many blurted about RaceControl..

**Comment by iebb** [comment]
Anyone wants to try my fork of F1 Web Viewer? I used some tricks to make logins and cookies work
Website is located at https://tv.ieb.systems/ , and the code is at https://github.com/iebb/F1-Web-Viewer , which is a fork from previously-broken https://github.com/iebb/F1-Web-Viewer

**Comment by KroontjesPen** [comment]
@iebb 
Request failed with status code 500
NOTE: This login involves processing of reese84 cookie, which is experimental and may break at any time. [F1 Web Viewer](https://github.com/bootsie123/F1-Web-Viewer) GitHub repository

**Comment by Tanguille** [comment]
> You can also allow manual token as temporary fix this weekend. Here is an easy way to get the current logged in session token from browser. 


@eXhumer I got the token. But what do I do with it? Doesn't work if I use it as password unfortunately.

**Comment by Tanguille** [comment]
> Anyone wants to try my fork of F1 Web Viewer? I used some tricks to make logins and cookies work Website is located at https://tv.ieb.systems/ , and the code is at https://github.com/iebb/F1-Web-Viewer , which is a fork from previously-broken https://github.com/iebb/F1-Web-Viewer

I tried your fork but got the following error: 

> Request failed with status code 500

**Comment by robvdpol** [comment]
If you have the subscription token you can paste it in the settings JSON file located at _%LOCALAPPDATA%\RaceControl\RaceControl.settings.json_ (field 'SubscriptionToken'). Make sure to set the 'LastLogin' to something recent (less than a week ago). Then it should auto login, but still you would have the DRM issue.

**Comment by FlyingWraptor** [comment]
@robvdpol Try requesting the playback urls from the v1 play endpoint instead of the v2 that should return the non-widevine version still

**Comment by PostMalone** [comment]
> If you have the subscription token you can paste it in the settings JSON file located at _%LOCALAPPDATA%\RaceControl\RaceControl.settings.json_ (field 'SubscriptionToken'). Make sure to set the 'LastLogin' to something recent (less than a week ago). Then it should auto login, but still you would have the DRM issue.

Not sure about the "LastLogin" part, am i supposed to put in the date there? 

Like this?: https://i.imgur.com/kS4SON8.png

**Comment by robvdpol** [comment]
> > If you have the subscription token you can paste it in the settings JSON file located at _%LOCALAPPDATA%\RaceControl\RaceControl.settings.json_ (field 'SubscriptionToken'). Make sure to set the 'LastLogin' to something recent (less than a week ago). Then it should auto login, but still you would have the DRM issue.
> 
> Not sure about the "LastLogin" part, am i supposed to put in the date there?
> 
> Like this?: https://i.imgur.com/kS4SON8.png

Put it like this:

`"LastLogin": "2022-04-08T13:17:23.2004797Z",`

**Comment by robvdpol** [comment]
> @robvdpol Try requesting the playback urls from the v1 play endpoint instead of the v2 that should return the non-widevine version still

Can you give an example URL? I'm having a hard time getting v1 to work for me. Which stream type are you using (for example BIG_SCREEN_HLS)?

**Comment by FlyingWraptor** [comment]
> ![image](https://user-images.githubusercontent.com/38633741/162445331-25f975a6-7799-4bac-902d-54d4b0c023d1.png)

Sometimes it does return m3u8 sometimes it returns mpeg-dash, as far as I've experienced it doesn't include widevine in either.

**Comment by dhuit90** [comment]
https://f1tv.formula1.com/1.0/R/NLD/WEB_HLS/ALL/CONTENT/PLAY?channelId=&contentId=1000005102

**Comment by robvdpol** [comment]
When I use that I get 403 Forbidden for the tokenised URL that is generated.

**Comment by iebb** [comment]
> When I use that I get 403 Forbidden for the tokenised URL that is generated.

only v2.0 is working apparently, 1.0 just gives 403

**Comment by FlyingWraptor** [comment]
> When I use that I get 403 Forbidden for the tokenised URL that is generated.

I might be mistaken, but I think you need to take the cookies in the response of the play request, and add those to all fragment & manifest requests:
![image](https://user-images.githubusercontent.com/38633741/162449123-4306b847-e4f6-46a7-bf5a-9bb1f6b3e670.png)
![image](https://user-images.githubusercontent.com/38633741/162449218-f84b7eec-db6c-4ace-bacd-a0700e62cec2.png)

**Comment by ricky732** [comment]
Guys, are there concrete chances that we can find a solution for quali and race? Even a provisional one..

**Comment by dhuit90** [comment]
> Guys, are there concrete chances that we can find a solution for quali and race? Even a provisional one..

I think using official sources is the way to go atm, this is pretty complex stuff and don't forget all the 3rd party devs to this in their free time

**Comment by iebb** [comment]
> Ok, this will require some more work. It basically means abandoning the Flyleaf video library and replacing it with an Edge WebView2 browser control that runs VideoJS. I have messed around with this in the past, and it should be able to support Widevine DRM. But it will take some time to implement, so please be patient.

@robvdpol Here is my working example, Australian GP is working fine using video.js and eme plugin.

https://github.com/iebb/F1-Web-Viewer/blob/master/src/components/Feed.vue#L128

Demo at https://live.sox.pm/

**Comment by KroontjesPen** [comment]
> > Ok, this will require some more work. It basically means abandoning the Flyleaf video library and replacing it with an Edge WebView2 browser control that runs VideoJS. I have messed around with this in the past, and it should be able to support Widevine DRM. But it will take some time to implement, so please be patient.
> 
> @robvdpol Here is my working example, Australian GP is working fine using video.js and eme plugin.
> 
> https://github.com/iebb/F1-Web-Viewer/blob/master/src/components/Feed.vue#L128
> 
> Demo at https://live.sox.pm/

Gives here in the Netherlands still:
 Request failed with status code 500

**Comment by ricky732** [comment]
Yeah same error here

**Comment by iebb** [comment]
> 

also @Tanguille 

this should work now, I forgot to update the backend when I modified cookie stuffs.
This website and https://tv.ieb.systems/ (US) and https://f1tv-eu.ieb.systems/ (DE) should all be working now.
(ieb.systems site are more preferred, while sox.pm is more like experimental playgrounds)

**Comment by KroontjesPen** [comment]
TOP
 https://f1tv-eu.ieb.systems/ (DE) **works**.

**Comment by PostMalone** [comment]
@iebb 

This is legendary stuff! 

Works like a charm here. Only thing would be to get this into an external player (MPV or similar). Other than that. Top stuff.

I hope this does not waste any of your server bandwidth, or does it?

**Comment by ricky732** [comment]
Ok so here it works, but it buffers a lot... Any chance to get a link for any external player that has not the DRM?

**Comment by iebb** [comment]
> @iebb
> 
> This is legendary stuff!
> 
> Works like a charm here. Only thing would be to get this into an external player (MPV or similar). Other than that. Top stuff.
> 
> I hope this does not waste any of your server bandwidth, or does it?

sox.pm is 100Mbps unmetered dedi box, which would allow ~20 simutaneous streams, while ieb.systems one are hosted on deta.sh for free (with some limitations such as not supporting 1080p)

**Comment by iebb** [comment]
> Ok so here it works, but it buffers a lot... Any chance to get a link for any external player that has not the DRM?

since sox.pm is hosted on < $15/m servers, and ieb.systems is for free that's pretty acceptable. also you can self-host it so that it depends on your own connection only. would provide a tutorial later, if everything works well

**Comment by ricky732** [comment]
Ok thank you!!!! I have a 30 mbps, it supported 4 multiple streams without problems if i gave it a bit of cache... Any tutorial about the syncro of feeds?

**Comment by iebb** [comment]
> Ok thank you!!!! I have a 30 mbps, it supported 4 multiple streams without problems if i gave it a bit of cache... Any tutorial about the syncro of feeds?

for replays there's a clock button, and for live streams clicking on LIVE should get you to the latest point?

**Comment by NH-Networks** [comment]
> Ok so here it works, but it buffers a lot... Any chance to get a link for any external player that has not the DRM?

will never work on a player without DRM, unless someone makes a "proxy" to hls

**Comment by ricky732** [comment]
Oh ok!! What is the link of sox.pm website? I was using an hls player firefox extension & i was very very happy with that.. What a pity I cannot use it anymore!

**Comment by iebb** [comment]
> 

live.sox.pm - however it's more like a tesing playground, which might explode at any time :D

**Comment by ricky732** [comment]
Doesn't load lol..

**Comment by koningcool** [comment]
Is there already a fix for the drm and the cookie issue (there is the manualy token fix for the cookie, but that is temporary)

**Comment by NH-Networks** [comment]
> Oh ok!! What is the link of sox.pm website? I was using an hls player firefox extension & i was very very happy with that.. What a pity I cannot use it anymore!

@iebb is working on self hosted version i see... maybe he can do some magic, time will tell

**Comment by ricky732** [comment]
That would be amazing

**Comment by iebb** [comment]
> > Oh ok!! What is the link of sox.pm website? I was using an hls player firefox extension & i was very very happy with that.. What a pity I cannot use it anymore!
> 
> @iebb is working on self hosted version i see... maybe he can do some magic, time will tell

It's a simple reverse proxy, all i need to do is some code cleanups, wouldn't take long

**Comment by sartsj** [comment]
> > i was able to login and continue using this app just fine, and the previous races from this and older seasons wor just fine. but i did notice that i was not able to watch Australian gp fp1 live or now as a archived session. other players stay empty but the internal one tries to load something, however that is just some pink mess with no audio. have they changed some drm things or is this just related to this issue as well and the upcoming 2.6.4 will resolve this issue as well? ![2022-04-08 (1)](https://user-images.githubusercontent.com/96973990/162370335-94b763a1-9c6e-4baf-97dc-de0319aa0b02.png)
> 
> As for this issue, for now you can still use the v1 endpoint to request a playback url, that one will return the original non-widevine hls stream format

Are you sure about this? I can also get unencrypted HLS on the v2 endpoint right now. It seems to me they just provide the unencrypted video when an event is not live anymore.

(assuming with v1 and v2 you mean just changing the /2.0/ to /1.0/ in the url and vice versa)

**Comment by FlyingWraptor** [comment]
> 



> > > i was able to login and continue using this app just fine, and the previous races from this and older seasons wor just fine. but i did notice that i was not able to watch Australian gp fp1 live or now as a archived session. other players stay empty but the internal one tries to load something, however that is just some pink mess with no audio. have they changed some drm things or is this just related to this issue as well and the upcoming 2.6.4 will resolve this issue as well? ![2022-04-08 (1)](https://user-images.githubusercontent.com/96973990/162370335-94b763a1-9c6e-4baf-97dc-de0319aa0b02.png)
> > 
> > 
> > As for this issue, for now you can still use the v1 endpoint to request a playback url, that one will return the original non-widevine hls stream format
> 
> Are you sure about this? I can also get unencrypted HLS on the v2 endpoint right now. It seems to me they just provide the unencrypted video when an event is not live anymore.
> 
> (assuming with v1 and v2 you mean just changing the /2.0/ to /1.0/ in the url and vice versa)

I must admit, I haven't tested the current weekend sessions, but as far as I am aware, last week I was able to circumvent the DRM using the 1.0/v1 endpoint

**Comment by sartsj** [comment]
> > 
> 
> > > > i was able to login and continue using this app just fine, and the previous races from this and older seasons wor just fine. but i did notice that i was not able to watch Australian gp fp1 live or now as a archived session. other players stay empty but the internal one tries to load something, however that is just some pink mess with no audio. have they changed some drm things or is this just related to this issue as well and the upcoming 2.6.4 will resolve this issue as well? ![2022-04-08 (1)]
> > > 
> > > 
> > > As for this issue, for now you can still use the v1 endpoint to request a playback url, that one will return the original non-widevine hls stream format
> > 
> > 
> > Are you sure about this? I can also get unencrypted HLS on the v2 endpoint right now. It seems to me they just provide the unencrypted video when an event is not live anymore.
> > (assuming with v1 and v2 you mean just changing the /2.0/ to /1.0/ in the url and vice versa)
> 
> I must admit, I haven't tested the current weekend sessions, but as far as I am aware, last week I was able to circumvent the DRM using the 1.0/v1 endpoint.
> 
> EDIT: Yes only live content is encrypted with DRM, replays work fine using both endpoints

Well last week it was also possible to get unencrypted streams from the v2 endpoint for live sessions :)
I guess we'll have to wait until FP3 to see whether the v1 trick works.

**Comment by ricky732** [comment]
@iebb so if I understood, if you make your code cleanups, we can host the stream by ourselves, but in that case, what player should we use to play it?

**Comment by iebb** [comment]
> 

still using the browser, the only difference is the bandwidth is completely your own

**Comment by Tanguille** [comment]
> 

I think they are overloaded because it keeps loading for me at the moment.

**Comment by iebb** [comment]
> > 
> 
> I think they are overloaded because it keeps loading for me at the moment.
> 
> Edit: BTW is there a thread on your repo or something which you can put a link for here so we can discuss it further without flooding this issue thread?

https://github.com/iebb/F1-Web-Viewer/issues/

**Comment by moi427** [comment]
> > 
> 
> @KroontjesPen , @Tanguille and @ricky732
> 
> this should work now, I forgot to update the backend when I modified cookie stuffs. This website and https://tv.ieb.systems/ (US) and https://f1tv-eu.ieb.systems/ (DE) should all be working now. (ieb.systems site are more preferred, while sox.pm is more like experimental playgrounds)

I'm sorry, but nothing is loading here after starting both the sites and login. I'm just a user of the great RaceControl, so most of the comments here are not something I understand.

**Comment by Digi421** [comment]
> > > If you have the subscription token you can paste it in the settings JSON file located at _%LOCALAPPDATA%\RaceControl\RaceControl.settings.json_ (field 'SubscriptionToken'). Make sure to set the 'LastLogin' to something recent (less than a week ago). Then it should auto login, but still you would have the DRM issue.
> > 
> > 
> > Not sure about the "LastLogin" part, am i supposed to put in the date there?
> > Like this?: https://i.imgur.com/kS4SON8.png
> 
> Put it like this:
> 
> `"LastLogin": "2022-04-08T13:17:23.2004797Z",`

This actually worked for me (Win10, app v2.6.3). Added Subscription Token (from browser) and LastLogin into RaceControl.settings.json. I can watch FP1 without problems but FP2 is garbled/encrypted -weird.
FP1: 
![image](https://user-images.githubusercontent.com/103273737/162480564-9b813122-f48f-4a64-bf93-91e47de5ec93.png)

FP2:
![image](https://user-images.githubusercontent.com/103273737/162480767-1ae57821-1cc2-4745-b759-c34c0af94bc8.png)

**Comment by thedave42** [comment]
Looks like v2 is returning unencrypted for FP1 now but still not for FP2.   I can get the unencrypted HLS for FP1 via the v2 api using BIG_SCREEN_HLS.  v1 is also now returning working, un-encrypted links for FP1 but still giving 403s for FP2.

**Comment by MueR** [comment]
Go figure, FOM skimping on having to pay the fees for the DRM..

**Comment by eXhumer** [comment]
> Go figure, FOM skimping on having to pay the fees for the DRM.

What makes you say that? The fact that they remove the DRM for you?

> Looks like v2 is returning unencrypted for FP1 now but still not for FP2. I can get the unencrypted HLS for FP1 via the v2 api using BIG_SCREEN_HLS. v1 is also now returning working, un-encrypted links for FP1 but still giving 403s for FP2.

> > 
> 
> > > > i was able to login and continue using this app just fine, and the previous races from this and older seasons wor just fine. but i did notice that i was not able to watch Australian gp fp1 live or now as a archived session. other players stay empty but the internal one tries to load something, however that is just some pink mess with no audio. have they changed some drm things or is this just related to this issue as well and the upcoming 2.6.4 will resolve this issue as well? ![2022-04-08 (1)](https://user-images.githubusercontent.com/96973990/162370335-94b763a1-9c6e-4baf-97dc-de0319aa0b02.png)
> > > 
> > > 
> > > As for this issue, for now you can still use the v1 endpoint to request a playback url, that one will return the original non-widevine hls stream format
> > 
> > 
> > Are you sure about this? I can also get unencrypted HLS on the v2 endpoint right now. It seems to me they just provide the unencrypted video when an event is not live anymore.
> > (assuming with v1 and v2 you mean just changing the /2.0/ to /1.0/ in the url and vice versa)
> 
> I must admit, I haven't tested the current weekend sessions, but as far as I am aware, last week I was able to circumvent the DRM using the 1.0/v1 endpoint.
> 
> EDIT: Yes only live content is encrypted with DRM, replays work fine using both endpoints

Using V1/V2 will have no difference to the actual stream they give you other than the errors. There is no bypass with V1. FOM protects live content and recent VODs with DRM for the first few hours and removes them to allow HLS playback like before. They have been doing this since Bahrain / Saudi Arabia.

@robvdpol I also suggest locking this issue until you have some update yourself, most of the comments here add nothing to help with the issue other than spreading FUD.

**Comment by DaveFlashNL** [comment]
weird, on Mac OS im just now watching FP1, and I was able to lift the m3u8 and dump it into ffmpeg and YouTube downloader, no drm in sight here. I thought they already added it no? oh one weird thing, YouTube_dl only saves the Portuguese audio and ffmpeg only the English commentary.

**Comment by iebb** [comment]
> weird, on Mac OS im just now watching FP1, and I was able to lift the m3u8 and dump it into ffmpeg and YouTube downloader, no drm in sight here. I thought they already added it no? oh one weird thing, YouTube_dl only saves the Portuguese audio and ffmpeg only the English commentary.

only the most recent VOD / live contents are with DRM (currently FP2 only)

**Comment by FlyingWraptor** [comment]
> 

I have not tested the last 2 sessions live sadly, but last weekend I had the same DRM issues, and my v1 fallback did work, adding the set-cookie headers to the headers of any http request to the cdn.

I'll test FP2 in a minute (as I heard people saying there's still DRM on playback) and FP3 in a couple of hours and provide you my findings.

**Comment by thedave42** [comment]
> I have not tested the last 2 sessions live sadly, but last weekend I had the same DRM issues, and my v1 fallback did work, adding the set-cookie headers to the headers of any http request to the cdn.

Can confirm I experienced this last weekend as well.  v1 would always return working URLs w/o DRM - even for live.  Last week the live was always an MPDs, even if you requested HLS, but the MPD was working and returning a stream in the clear.

**Comment by sartsj** [comment]
I 100% used v2 endpoint last weekend for live sessions and got unencrypted HLS streams. It also has to do with the user-agents provided what stream you get, so probably that it why you were getting encrypted streams.

**Comment by eXhumer** [comment]
`GET https://f1tv.formula1.com/1.0/R/ENG/WEB_HLS/ALL/CONTENT/PLAY?channelId=1033&contentId=1000005127`

Response
```
{"resultCode":"OK","message":"200","errorDescription":"","resultObj":{"entitlementToken":"entitlementToken","url":"https://ott-video-cf.formula1.com/f1d26a28c95485cc/out/v1/613432380c864ec6847609656c317695/index.mpd?kid=1042&exp=1649527434&ttl=1440&token=<token>&start=1649397304&end=1649401586","streamType":"DASHWV","drmType":"widevine","laURL":"https://f1tv.formula1.com/2.0/R/ENG/WEB_HLS/ALL/CONTENT/LA/widevine?channelId=1033&contentId=1000005127","channelId":1033},"systemTime":1649441033706}
```

**Comment by DaveFlashNL** [comment]
> only the most recent VOD / live contents are with DRM (currently FP2 only)

so after FP3, FP2 becomes drm free? that seems weird

**Comment by JohnLBergqvist** [comment]
For those who are struggling to understand why some sessions are encrypted and some aren't: F1 TV is now returning the DRM-protected DASH format for videos that originate as a live session. Other videos use HLS. Since the beginning of this week, live streams have been returned as DASH, even when you've used the HLS endpoints.

Most of the time (but not always), a few hours after the session, the original DRM-protected replay is replaced with a standalone video, which at that point if you request it, you'll get it in HLS format. They've done this for FP1, but at the time of writing, not FP2.

**Comment by DaveFlashNL** [comment]
They've done this for Australia FP1 (which is why it's viewable), but at the time of writing, not FP2.

i suspect either in a few hrs time or after fp3, they've processed the videos through their system and FP2 becomes available in HLS on that endpoint you mentions, based on the comments here I assume it it this way because F1 still needs the HLS streams for certain devices/scenarios, that they have yet to migrate to the new systems and endpoints that are fully dorm's up.

**Comment by robvdpol** [comment]
Login issue is fixed in 2.6.4 release.

**Comment by snowice0** [comment]
Looks like im running into the issue just before FP1 Miami
